File: tests/fixtures/java/classes.java
package fixtures.classes;

class Worker extends BaseWorker implements Runnable {
    private final Service service;

    public Worker(Service service) {
        this.service = service;
    }

    @Override
    public void run() {
        service.execute();
    }

    private static class State {
        boolean ready() {
            return check();
        }
    }
}

File: tests/fixtures/ruby/classes.rb
module Fixtures
  module Services
    class Worker < BaseWorker
      DEFAULT_LIMIT = build_limit()

      def initialize(client)
        @client = client
      end

      def run
        @client.call
      end

      def self.create
        new()
      end

      class << self
        def configure
          setup()
        end
      end
    end
  end
end
